packages/experience-builder-sdk/src/blocks/editor/VisualEditorContext.tsx

```diff
@@ -221,9 +227,10 @@ export function VisualEditorContextProvider({
             changedNode?: CompositionComponentNode;
             changedValueType?: CompositionComponentPropValue['type'];
           } = payload;
+          // Make sure to first store the design components before setting the tree and thus triggering a rerender
+          designComponents && setDesignComponents(designComponents);
```

Comment:
[subjective] 🙄 I also love the  hacky `&&`  pattern :) (and it works great for JSX), but my understanding was that for non-JSX code this should be used only on special occasions ;)

Reply:
You're right. It's not good - I just moved lines here without thinking. I will refactor this part :)